plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeStorage.java

```diff
@@ -0,0 +1,49 @@
+package io.trino.plugin.exchange;
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+
+import javax.crypto.SecretKey;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public interface FileSystemExchangeStorage
+        extends AutoCloseable
+{
+    void createDirectories(URI dir) throws IOException;
+
+    SliceInput getSliceInput(URI file, Optional<SecretKey> secretKey) throws IOException;
+
+    SliceOutput createSliceOutput(URI file, Optional<SecretKey> secretKey) throws IOException;
+
+    boolean exists(URI file);
+
+    void createFile(URI file) throws IOException;
```

Comment:
Yes but that will create unnecessary memory allocation (and get GC'ed later). Since the implementation is basically one-liner, I would prefer to avoid memory allocation in this method.

Reply:
Sounds good. Maybe rename it to `createEmptyFile`?